doc/source/serve/http-guide.md

```diff
@@ -95,88 +117,148 @@ def f():
-    @app.get("/subpath")
-    def method(self):
-        return "Hello 1!"
+You can specify different type signatures to facilitate the extraction of HTTP fields, including
+- [query parameters](https://fastapi.tiangolo.com/tutorial/query-params/),
+- [body parameters](https://fastapi.tiangolo.com/tutorial/body/),
+and 
+- [many other data types](https://fastapi.tiangolo.com/tutorial/extra-data-types/).
 
-@serve.deployment(route_prefix="/api2")
-@serve.ingress(app)
-class FastAPIWrapper2:
-    @app.get("/subpath")
-    def method(self):
-        return "Hello 2!"
+For more details, you can take a look at the [FastAPI documentation](https://fastapi.tiangolo.com/).
+
+You can use adapters in different scenarios within Serve, which we will go over one by one:
+
+- Ray AIR `Predictor`
+- Serve Deployment Graph `DAGDriver`
+- Embedded in Bring Your Own `FastAPI` Application
+
+### Ray AIR `Predictor`
+
+Ray Serve provides a suite of adapters to convert HTTP requests to ML inputs like `numpy` arrays.
+You can use them together with the [Ray AI Runtime (AIR) model wrapper](air-serving-guide) feature
+to one-click deploy pre-trained models.
+
+As an example, we provide a simple adapter for an *n*-dimensional array.
 
-FastAPIWrapper1.deploy()
-FastAPIWrapper2.deploy()
+When using [model wrappers](air-serving-guide), you can specify your HTTP adapter via the `http_adapter` field:
+
+```python
+from ray import serve
+from ray.serve.http_adapters import json_to_ndarray
+from ray.serve import PredictorDeployment
+
+serve.run(PredictorDeployment.options(name="my_model").bind(
+    my_ray_air_predictor,
+    my_ray_air_checkpoint,
+    http_adapter=json_to_ndarray
+))
 ```
 
-In this example, requests to both `/api1` and `/api2` would return `Hello from the root!` while a request to `/api1/subpath` would return `Hello 1!` and a request to `/api2/subpath` would return `Hello 2!`.
+### Serve Deployment Graph `DAGDriver`
 
-To try it out, save a code snippet in a local python file (i.e. main.py) and in the same directory, run the following commands to start a local Ray cluster on your machine.
+When using a [Serve Deployment Graph](serve-deployment-graph), you can configure
+`ray.serve.drivers.DAGDriver` to accept an HTTP adapter via its `http_adapter` field.
 
-```bash
-ray start --head
-python main.py
+For example, the `json_request` adapter parses JSON in the HTTP body:
+
+```python
+from ray.serve.drivers import DAGDriver
+from ray.serve.http_adapters import json_request
+from ray.dag.input_node import InputNode
+
+with InputNode() as input_node:
+    # ...
+    dag = DAGDriver.bind(other_node, http_adapter=json_request)
 ```
 
+### Embedded in your existing `FastAPI` Application
 
-### Configuring HTTP Server Locations
+You can also bring the adapter to your own FastAPI app using
+[Depends](https://fastapi.tiangolo.com/tutorial/dependencies/#import-depends).
+The input schema will automatically be part of the generated OpenAPI schema with FastAPI.
 
```

Comment:
The input schema automatically become part of the generated OpenAPI schema with FastAPI.

(just using present tense)